### PR TITLE
Fixes hub entry data collection

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Changes the hub to use the special `&#8212;` character instead of the `-` character. This means data collection will be handled more gracefully.

@Crossedfall will need to take a look at this to ensure that the <a href... tags have no effect.

This is what it will look like instead:

image

Closes #10912

## Changelog
:cl:
fix: Fixes the hub entry not using the correct character to divide the station name.
/:cl:

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
 		// Station name is going to be truncated with ...
 		if (discordurl)
 			if (server_name)
-				s += "<a href='[discordurl]'><b>[server_name] - [copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
+				s += "<a href='[discordurl]'><b>[server_name]&#8212;[copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<a href='[discordurl]'><b>[server_name] &#8212; [copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
```

File: code/game/world.dm [hunk 2]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<a href='[discordurl]'><b>[copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
 		else
 			if (server_name)
-				s += "<b>[server_name] - [copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
+				s += "<b>[server_name]&#8212;[copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<b>[server_name] &#8212; [copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
```

File: code/game/world.dm [hunk 3]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<b>[copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
 	else
 		// Station name can be displayed in full
 		if (discordurl)
 			if (server_name)
-				s += "<a href='[discordurl]'><b>[server_name] - [station_name]</b></a><br>"
+				s += "<a href='[discordurl]'><b>[server_name]&#8212;[station_name]</b></a><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<a href='[discordurl]'><b>[server_name] &#8212; [station_name]</b></a><br>"
```

File: code/game/world.dm [hunk 4]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<a href='[discordurl]'><b>[station_name]</b></a><br>"
 		else
 			if (server_name)
-				s += "<b>[server_name] - [station_name]</b><br>"
+				s += "<b>[server_name]&#8212;[station_name]</b><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<b>[server_name] &#8212; [station_name]</b><br>"
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]

**Comment by PowerfulBacon** [comment]
Hold on, let me update the calculation to include the spaces (otherwise it doesn't know when to cut the station name)

File: code/game/world.dm [hunk 5]
```diff
@@ -372,7 +372,7 @@ GLOBAL_VAR(restart_counter)
 		popcaptext = "/[popcap]"
 
 	// Determine our character usage
-	var/character_usage = 86	// Base character usage
+	var/character_usage = 90	// Base character usage
```
Comment (PowerfulBacon) [inline code/game/world.dm]:
```suggestion
	var/character_usage = 92	// Base character usage
```

**Comment by PowerfulBacon** [comment]
Ok, should be good now

File: code/game/world.dm [hunk 6]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
 		// Station name is going to be truncated with ...
 		if (discordurl)
 			if (server_name)
-				s += "<a href='[discordurl]'><b>[server_name] - [copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
+				s += "<a href='[discordurl]'><b>[server_name] &#8212; [copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<a href='[discordurl]'><b>[server_name]</b> - <b>[copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
```

File: code/game/world.dm [hunk 7]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<a href='[discordurl]'><b>[copytext(station_name, 1, station_name_limit - 3)]...</b></a><br>"
 		else
 			if (server_name)
-				s += "<b>[server_name] - [copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
+				s += "<b>[server_name] &#8212; [copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<b>[server_name]</b> - <b>[copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
```

File: code/game/world.dm [hunk 8]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<b>[copytext(station_name, 1, station_name_limit - 3)]...</b><br>"
 	else
 		// Station name can be displayed in full
 		if (discordurl)
 			if (server_name)
-				s += "<a href='[discordurl]'><b>[server_name] - [station_name]</b></a><br>"
+				s += "<a href='[discordurl]'><b>[server_name] &#8212; [station_name]</b></a><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<a href='[discordurl]'><b>[server_name]</b> - <b>[station_name]</b></a><br>"
```

File: code/game/world.dm [hunk 9]
```diff
@@ -399,24 +399,24 @@ GLOBAL_VAR(restart_counter)
...
 			else
 				s += "<a href='[discordurl]'><b>[station_name]</b></a><br>"
 		else
 			if (server_name)
-				s += "<b>[server_name] - [station_name]</b><br>"
+				s += "<b>[server_name] &#8212; [station_name]</b><br>"
```
Comment (Crossedfall) [inline code/game/world.dm]:
```suggestion
				s += "<b>[server_name]</b> - <b>[station_name]</b><br>"
```

**Review by Crossedfall — COMMENTED** [review]
It's the \</b> that matters, not the special character. Character usage count shouldn't change I believe

**Review by Crossedfall — APPROVED** [review]

**Comment by Crossedfall** [comment]
I'm good with it now. I verified that my data collection and other 3rd party collectors all appear to be functioning again. I'll merge it tomorrow to give time for any last minute changes.

We could probably adjust the character usage a bit 
image

**Pull request merged**